src/server/loadtest/loadtest-middleware.js

```diff
@@ -0,0 +1,73 @@
+// @flow
+import jwt from 'jsonwebtoken'
+import passport from 'passport'
+import { ExtractJwt, Strategy } from 'passport-jwt'
+import { Router } from 'express'
+import * as ethUtil from 'ethereumjs-util'
+import { get, defaults } from 'lodash'
+import logger from '../../imports/pino-logger'
+import { wrapAsync, onlyInEnv } from '../utils/helpers'
+import UserDBPrivate from '../db/mongo/user-privat-provider'
+import Config from '../server.config.js'
+import AdminWallet from '../blockchain/AdminWallet'
+
+const jwtOptions = {}
+jwtOptions.jwtFromRequest = ExtractJwt.fromAuthHeaderAsBearerToken()
+jwtOptions.secretOrKey = Config.jwtPassword
+
+export const strategy = new Strategy(jwtOptions, async (jwtPayload, next) => {
+  const log = logger.child({ from: 'loadtest-middleware' })
+  // usually this would be a database call:
+  let user = await UserDBPrivate.getUser(jwtPayload.loggedInAs)
+  log.debug('payload received', { jwtPayload, user })
+  //if user is empty make sure we have something
+  user = defaults(user, jwtPayload, { identifier: jwtPayload.loggedInAs })
+  if (get(jwtPayload, 'loggedInAs')) {
+    next(null, user)
+  } else {
+    next(null, false)
+  }
+})
+
+const recoverPublickey = (signature, msg, nonce) => {
+  const sig = ethUtil.fromRpcSig(signature)
+
+  const messageHash = ethUtil.keccak(
+    `\u0019Ethereum Signed Message:\n${(msg.length + nonce.length).toString()}${msg}${nonce}`
+  )
+
+  const publicKey = ethUtil.ecrecover(messageHash, sig.v, sig.r, sig.s)
+  const recovered = ethUtil.bufferToHex(ethUtil.pubToAddress(publicKey))
+  return recovered
+}
+
+const setup = (app: Router) => {
+  passport.use(strategy)
+
+  app.use(passport.initialize())
+
```

Comment:
up to here is all uncessaray duplicated code.
please see how other middlewares are written

Reply:
[done](https://github.com/GoodDollar/GoodServer/pull/72/commits/e9a0bd5781aca9f1d3f71feec2205c52b45e4210)